pkg/controllers/provisioning/scheduling/nodeset.go

```diff
@@ -28,14 +32,20 @@ import (
 )
 
 type NodeSet struct {
-	client          client.Client
+	kubeClient      client.Client
 	daemonResources v1.ResourceList
 	nodes           []*Node
+	constraints     *v1alpha5.Constraints
+	instanceTypes   []cloudprovider.InstanceType
+	topology        *Topology
 }
 
-func NewNodeSet(ctx context.Context, constraints *v1alpha5.Constraints, client client.Client) (*NodeSet, error) {
+func NewNodeSet(ctx context.Context, topology *Topology, constraints *v1alpha5.Constraints, instanceTypes []cloudprovider.InstanceType, client client.Client) (*NodeSet, error) {
```

Comment:
The responsibilities of this class overlaps pretty heavily with Scheduler, and both are pretty small.

For example, 4 of the 6 member variables are injected from topology, and the other two could simply live on the stack. 

I'll see if I can cut you a PR.